runtime/src/bank.rs

```diff
@@ -3073,9 +3073,16 @@ impl Bank {
     }
 
     pub fn get_fee_for_message(&self, message: &SanitizedMessage) -> Option<u64> {
-        let blockhash_queue = self.blockhash_queue.read().unwrap();
-        let lamports_per_signature =
-            blockhash_queue.get_lamports_per_signature(message.recent_blockhash())?;
+        let lamports_per_signature = match self.check_message_for_nonce(message) {
```

Comment:
Was trying to avoid loading the nonce account for each message fee calculation, but I guess that's unavoidable if we want to support nonces here.
Would help if we checked the blockhash queue first, and then fell back to loading and parsing an account if not found.

Reply:
Ah yes, good point.  I'll flip that around

Reply:
done